docs/src/tutorial/parallelization.rst

```diff
@@ -0,0 +1,319 @@
+code in between.
+
+.. tabs::
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/normalize.pyx
+        
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/normalize.py
+
+The C code is approximately:
+
+.. code-block:: C
+
+    #pragma omp parallel private(norm) reduction(+:total)
+    {
+        /* some calculations of array size... */
+        #pragma omp for firstprivate(i) lastprivate(i)
+        for (__pyx_t_2 = 0; __pyx_t_2 < __pyx_t_3; __pyx_t_2++){
+            /* ... */
+        }
+        norm = sqrt(total);
+        #pragma omp for firstprivate(i) lastprivate(i)
+        for (__pyx_t_2 = 0; __pyx_t_2 < __pyx_t_3; __pyx_t_2++){
+            /* ... */
+        }
+    }
+
+Allocating "scratch space" for each thread
+------------------------------------------
+
+Suppose that each thread requires a small amount of scratch space
+to work in.  They cannot share scratch space because that would
+lead to data races.  In this case the allocation and deallocation
+is done in a parallel section (so occurs on a per-thread basis)
+surrounding a loop which then uses the scratch space.
+
+Our example here uses C++ to find the median of each column in
+a 2D array (just a parallel version of ``numpy.median(x, axis=0)``).
+We must reorder each column to find the median of it, but don't want
+to modify the input array.  Therefore, we allocate a C++ vector per
+thread to use as scratch space, and work in that.  For efficiency
+the vector is allocated outside the ``prange`` loop.
+
+.. tabs::
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/median.pyx
+        
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/median.py
+
+In the generated code the ``scratch`` variable is marked as
+``private`` in the outer parallel block.  A rough outline is:
+
+.. code-block:: C++
+
+    #pragma omp parallel private(scratch)
+    {
+        scratch = new std::vector<double> ((x.shape[0]))
+        #pragma omp for firstprivate(i) lastprivate(i) lastprivate(j) lastprivate(median_it)
+        for (__pyx_t_9 = 0; __pyx_t_9 < __pyx_t_10; __pyx_t_9++){
+            i = __pyx_t_9;
+            /* implementation goes here */
+        }
+        /* some exception handling detail omitted */
+        delete scratch;
+    }
+
+Performing different tasks on each thread
+-----------------------------------------
+
+Finally, if you manually specify the number of threads and
+then identify each thread using ``omp.get_thread_num()``
+you can manually split work between threads.  This is
+a fairly rare use-case in Cython, and probably suggests
+that the ``threading`` module is more suitable for what
+you're trying to do.  However it is an option.
+
+.. tabs::
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/manual_work.pyx
```

Comment:
```suggestion
        .. literalinclude:: ../../examples/tutorial/parallelization/manual_work.pyx
           :lines: 2-
```

Reply:
This will remove tag comment connected to CI which should not bother consumer of documentation.